sisu-osgi/sisu-osgi-connect/src/main/java/org/eclipse/sisu/osgi/connect/PlexusFrameworkConnectServiceFactory.java

```diff
@@ -129,23 +131,22 @@ synchronized PlexusConnectFramework getFramework(ClassRealm realm) throws Bundle
 		p.put(Constants.FRAMEWORK_STORAGE, storagePath + File.separator + "storage");
 		p.put(Constants.FRAMEWORK_BEGINNING_STARTLEVEL, "6");
 		p.put("osgi.instance.area", storagePath + File.separator + "instance");
-		ServiceLoader<ConnectFrameworkFactory> sl = ServiceLoader.load(ConnectFrameworkFactory.class,
-				getClass().getClassLoader());
-		ConnectFrameworkFactory factory = sl.iterator().next();
+
+		var loader = ServiceLoader.load(ConnectFrameworkFactory.class, getClass().getClassLoader());
+		ConnectFrameworkFactory factory = loader.findFirst().orElseThrow();
```

Comment:
Please add a meaningful exception (with message) to be thrown here

Reply:
Done.